### PR TITLE
[CAZ-2046] Add proxies for Clean Air Zones, Compliance and Vehicle Details

### PR DESCRIPTION
**Pull request by cruddasj**
Adds a clean air zone proxy to v1/payments/clean-air-zones. Note the hyphen use here is to mirror VCCS for consistency in endpoint naming. The Clean Air Zone Name Getter service has been abstracted to just a Clean Air Zone service, as the same repository tiers are applicable.

Use of common lib for CAZ definition covered in VCCS under InformedSolutions/JAQU-CAZ-Vehicle-Compliance-Checker-API@a5fb14b

This further adds v1/payments/compliance. DTO switch to common lib in VCCS formed part of the same commit linked above (see https://github.com/InformedSolutions/JAQU-CAZ-Vehicle-Compliance-Checker-API/blob/ca419db82782a676fc470981ef3b6c192e7fdcab/src/main/java/uk/gov/caz/vcc/controller/VehicleController.java#L16 for reference on specific compliance outcome DT object).

### CODE REVIEW

**Comment by cruddasj** [comment]
That last set of commits drops code smells by 7 (to 22) and tech debt from 3h to 2h in sonar. No new issues introduced under new code 👍

File: src/main/java/uk/gov/caz/psr/service/CleanAirZoneService.java [hunk 1]
```diff
@@ -18,10 +19,32 @@
...
-public class CleanAirZoneNameGetterService {
+public class CleanAirZoneService {
 
   private final VccsRepository vccsRepository;
 
+  /**
+   * Gets clean air zones from VCCS.
+   *
+   * @return {@link CleanAirZonesResponse} A list of parsed Clean Air Zones
+   * @throws NullPointerException if {@code cleanAirZoneId} is null
```
Comment (plenza) [inline src/main/java/uk/gov/caz/psr/service/CleanAirZoneService.java]: `@throws` clauses are invalid in this context - they won't happen

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -158,6 +158,18 @@ swagger:
         paid-dates: A list of days which are already paid.
       paid-payments-response:
         results: Collection of VRNs and dates which are already paid in a given date range.
+      cleanAirZones:
+        cleanAirZoneDetails: List of Clean Air Zones
+      cleanAirZone:
+        cleanAirZoneId: System ID of the clean air zone
+        name: Name of Tariff
```
Comment (plenza) [inline src/main/resources/application.yml]: Tariff?

File: src/main/java/uk/gov/caz/psr/service/CleanAirZoneService.java (hunk 1 above)
Comment (cruddasj) [reply src/main/java/uk/gov/caz/psr/service/CleanAirZoneService.java]: Sorted 👍 - as a brief side note, I've drawn the method signature from the fetch(UUID cleanAirZoneId) signature. I'm going to remove the throws declaration from that too (given the exception being raised is not possible in the context).

File: src/main/java/uk/gov/caz/psr/service/VehicleComplianceRetrievalService.java [hunk 3]
```diff
@@ -77,6 +79,28 @@
...
+   * @throws HttpException http tier exception encountered when invoking VCCS
+   */
+  public ComplianceResultsDto retrieveVehicleCompliance(String vrn,
+      String zones) {
+    try {
+      Response<ComplianceResultsDto> response = vccsRepository.findComplianceSync(vrn, zones);
+      return response.body();
+    } catch (HttpException e) {
+      log.error("Failed to retrieve vehicle compliance resul from VCCS");
+      throw e;
```
Comment (kingarj) [inline src/main/java/uk/gov/caz/psr/service/VehicleComplianceRetrievalService.java]: Should this throw an external service exception?

**Comment by cruddasj** [comment]
That last commit changes things quite a bit to support passthrough of unhappy path error codes.

**Comment by cruddasj** [comment]
Merging down following local testing and test coverage remaining at 100%

**Pull request merged**